src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -152,27 +155,183 @@ public void AddPackageReference(string projectPath, LibraryDependency libraryDep
+        /// Add package name and version into the props file.
+        /// </summary>
+        /// <param name="itemGroup">Item group that needs to be modified in the props file.</param>
+        /// <param name="libraryDependency">Package Dependency of the package to be added.</param>
+        private void AddPackageVersionIntoPropsItemGroup(ProjectItemGroupElement itemGroup,
+            LibraryDependency libraryDependency)
+        {
+            // Add both package reference information and version metadata using the PACKAGE_VERSION_TYPE_TAG.
+            var item = itemGroup.AddItem(PACKAGE_VERSION_TYPE_TAG, libraryDependency.Name);
+            ProjectMetadataElement versionAttribute = null;
+            var packageVersion = AddVersionMetadata(libraryDependency, item, versionAttribute);
+
+            Logger.LogInformation(string.Format(CultureInfo.CurrentCulture,
+            Strings.Info_AddPkgAdded,
+            packageVersion,
+            libraryDependency.Name,
+            itemGroup.ContainingProject.FullPath));
+
+            AddExtraMetadata(libraryDependency, item);
+        }
+
+        /// <summary>
+        /// Add package name and version into item group.
+        /// </summary>
+        /// <param name="itemGroup">Item group to add to.</param>
+        /// <param name="libraryDependency">Package Dependency of the package to be added.</param>
+        private void AddPackageReferenceIntoItemGroup(ProjectItemGroupElement itemGroup,
+            LibraryDependency libraryDependency)
+        {
+            // Add both package reference information and version metadata using the PACKAGE_REFERENCE_TYPE_TAG.
+            var item = itemGroup.AddItem(PACKAGE_REFERENCE_TYPE_TAG, libraryDependency.Name);
+            ProjectMetadataElement versionAttribute = null;
+            var packageVersion = AddVersionMetadata(libraryDependency, item, versionAttribute);
+
+            Logger.LogInformation(string.Format(CultureInfo.CurrentCulture,
+            Strings.Info_AddPkgAdded,
+            packageVersion,
+            libraryDependency.Name,
+            itemGroup.ContainingProject.FullPath));
+
+            AddExtraMetadata(libraryDependency, item);
+        }
+
+        /// <summary>
+        /// Add only the package name into the project file for projects onboarded to CPM.
+        /// </summary>
+        /// <param name="itemGroup">Item group to add to.</param>
+        /// <param name="libraryDependency">Package Dependency of the package to be added.</param>
+        private void AddPackageReferenceIntoItemGroupCPM(ProjectItemGroupElement itemGroup,
+            LibraryDependency libraryDependency)
+        {
+            // Only add the package reference information using the PACKAGE_REFERENCE_TYPE_TAG.
+            ProjectItemElement item = itemGroup.AddItem(PACKAGE_REFERENCE_TYPE_TAG, libraryDependency.Name);
+            string packageVersion = "No package version added because this project is onboarded to CPM";
```

Comment:
Consider adding strings to the resources file so that they can be localized. We can add a new entry in the resource file for CPM projects. For example, `info : Adding PackageReference for package 'Newtonsoft.Json' into project 'C:\Users\user1\source\repos\ConsoleApp9\ConsoleApp9\ConsoleApp9.csproj' and Version into the Directory.Packages.Props file`